### PR TITLE
Add CI build workflow

### PR DESCRIPTION
**Pull request by Alexander-Ploskin**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (TimPushkin) [inline .github/workflows/build.yml]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 17
+          java-package: jdk
+          architecture: x64
```
Comment (TimPushkin) [inline .github/workflows/build.yml]:
```suggestion
      - uses: actions/setup-java@v3
        with:
          distribution: 'temurin'
          java-version: '17'
          cache: 'gradle'
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle
```
Comment (TimPushkin) [inline .github/workflows/build.yml]: This can be omitted when using `actions/setup-java@v3`

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Build
+        run: ./gradlew :clean :assemble
```
Comment (TimPushkin) [inline .github/workflows/build.yml]: I don't think you need `:clean` since workflows start in a new environment every time

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          java-package: jdk
```
Comment (TimPushkin) [inline .github/workflows/build.yml]: Why do you specify this if it is the default?

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+          cache: gradle
+          distribution: corretto
+      - name: Build
+        run: ./gradlew :clean :assemble
+      - name: Cleanup Gradle Cache
+        # Remove some files from the Gradle cache, so they aren't cached by GitHub Actions.
+        # Restoring these files from a GitHub Actions cache might cause problems for future builds.
+        run: |
+          rm -f ~/.gradle/caches/modules-2/modules-2.lock
+          rm -f ~/.gradle/caches/modules-2/gc.properties
```
Comment (TimPushkin) [inline .github/workflows/build.yml]: This can be omitted when using `actions/setup-java@v3`

File: .github/workflows/build.yml (hunk 4 above)
Comment (TimPushkin) [reply .github/workflows/build.yml]: Also, take a look at [this](https://github.com/gradle/gradle-build-action) -- I've never used it, but GitHub Actions Gradle template suggests it

**Comment by TimPushkin** [comment]
When using `gradle-build-action`, you should be able to run Gradle with it directly: [[this](https://github.com/gradle/gradle-build-action)](https://github.com/gradle/gradle-build-action#gradle-execution)

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: corretto
+      - name: Build
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: build
+          gradle-executable: ./gradlew
```
Comment (TimPushkin) [inline .github/workflows/build.yml]: Why is this needed? It is said [here](https://github.com/gradle/gradle-build-action#using-a-specific-gradle-executable) that this executable should be the default

**Review by DmiitriyJarosh — APPROVED** [review]

**Review by TimPushkin — APPROVED** [review]
